### PR TITLE
[AIRFLOW-3516] Support to create k8 worker pods in batches

### PR DESCRIPTION
**Pull request by ramandumcs**
Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,6 +109,7 @@ Currently **officially** using Airflow:
 1. [90 Seconds](https://90seconds.tv/) [[@aaronmak](https://github.com/aaronmak)]
 1. [99](https://99taxis.com) [[@fbenevides](https://github.com/fbenevides), [@gustavoamigo](https://github.com/gustavoamigo) & [@mmmaia](https://github.com/mmmaia)]
 1. [AdBOOST](https://www.adboost.sk) [[AdBOOST](https://github.com/AdBOOST)]
+1. [Adobe](https://www.adobe.com/)
```
Comment (OmerJog) [inline README.md]: This isn't related to this PR.
Please open a new PR for this.

Also, once removing it please squash your commits

**Comment by feng-tao** [comment]
PTAL @dimberman

File: README.md (hunk 1 above)
Comment (ramandumcs) [reply README.md]: Thanks @OmerJog. Updated the PR

**Comment by dimberman** [comment]
@feng-tao apologies for the delay, getting over a cold. Will check out now.

**Comment by dimberman** [comment]
@ramandumcs What is the use-case where a user would want to launch in batches rather than greedily launch tasks as they're received?

**Comment by ramandumcs** [comment]
Thanks @dimberman for looking in to this PR.

As per current implementation K8 Executor submits/creates one k8 worker pod per scheduler loop. 
Scheduler creates a single k8 worker pod inside self.executor.heartbeat() function of jobs.py file. In self.executor.heartbeat() function it calls self.sync() which submits/creates single k8 worker pod. 
task_queue might have 100s of tasks to be run but only one task gets submitted per scheduler loop which increases the task scheduling latency/delay.

Each scheduling loop takes a minimum of 1 second so scheduling latency of last task in the task queue with 1000 tasks will be atleast 1000 seconds.
(We have also observed scheuling loop sometimes taking ~ 2 to 3 seconds which further increases the task latency)

We have a use case to run 1000s of concurrent tasks and we started using Airflow with K8 executor where we observed and investigated this behaviour.
So we are proposing a fix to have some control over the number of worker pods that get submitted per loop. Ideal scenarion might haven been to submit all the tasks  per loop but it can impact the scheduler's processing of DAG files as these are synchronous call. 
So we made this configurable.
Please let us know if it makes sense and please also share your thoughts/comments

**Comment by dimberman** [comment]
@ramandumcs ohh ok that makes total sense. This LGTM. @kaxil is it too late to put this in 1.10.2?

**Comment by dimberman** [comment]
@feng-tao good to go on my end!

**Comment by feng-tao** [comment]
thanks @dimberman , @ramandumcs , couples of qqs for my understanding:
1. do we need to  add the config in default_test.cfg?
2. is there a way to test the behavior? 
3. My understanding for the change is that previously we launch one pod per single heartbeat call, now  we could launch multiple pods(not with multithreading) inside single heartbeat call?

**Comment by ramandumcs** [comment]
Yup, Now we are launching multiple pods per heartbeat call. I will see if we can add test cases for this particular behavior

**Pull request merged**